internal/codegen/blocks.go

```diff
@@ -33,6 +41,45 @@ func parseBlocks(filePath string) (map[string]string, error) {
 	for i := 0; scanner.Scan(); i++ {
 		line := scanner.Text()
 		matches := blockPattern.FindStringSubmatch(line)
+		if len(matches) == 0 {
+			// 0: full match
+			// 1: comment prefix
+			// 2: / if end of block
+			// 3: block name
+			// 4: block args, if present
+			v2Matches := v2BlockPattern.FindStringSubmatch(line)
+			if len(v2Matches) == 5 {
+				cmd := v2Matches[3]
+				if v2Matches[2] == "/" {
+					if cmd == endStatement {
+						return nil, fmt.Errorf("line %d: Stencil::EndBlock with a <</, should use <</Stencil::Block>> instead", i+1)
+					}
+
+					// If there is a /, it's a closing tag and we should
+					// translate it to a closing block command
+					cmd = endStatement
+					if v2Matches[4] != "" {
+						return nil, fmt.Errorf("line %d: expected no arguments to <</Stencil::Block>>", i+1)
+					}
+
+					v2Matches[4] = fmt.Sprintf("(%s)", curBlockName)
+				} else if cmd == endStatement {
+					// If it's not a closing tag, but the command is EndBlock,
+					// we should error. This is because we don't want to
+					// allow users to use the old EndBlock command
+					// without a closing tag
+					return nil, errors.Errorf("line %d: EndBlock command must be used with a closing tag", i+1)
+				}
+
+				// fake the old matches format so we can reuse the same code
+				matches = []string{
+					v2Matches[0],
+					v2Matches[1],
```

Comment:
Are you using this capture at all? Seems like you could uncapture it if you were feeling tidy.

Reply:
Which one? It's not super clear from the comment here which it's referring too 😢

Reply:
The first capture group being pulled out of `v2matches`. You're capturing the comment leader but never referencing it, AFAICT?